lib/asserts.nix

```
@@ -2,78 +2,122 @@
 
 rec {
 
-  /* Throw if pred is false, else return pred.
-     Intended to be used to augment asserts with helpful error messages.
+  /**
+    Throw if pred is false, else return pred.
+    Intended to be used to augment asserts with helpful error messages.
 
-     Example:
-       assertMsg false "nope"
-       stderr> error: nope
+    # Example
 
-       assert assertMsg ("foo" == "bar") "foo is not bar, silly"; ""
-       stderr> error: foo is not bar, silly
+    ```nix
+    assertMsg false "nope"
+    stderr> error: nope
+    assert assertMsg ("foo" == "bar") "foo is not bar, silly"; ""
+    stderr> error: foo is not bar, silly
+    ```
+
+    # Type
+
+    ```
+    assertMsg :: Bool -> String -> Bool
+    ```
+
+    # Arguments
+
+    `pred`
+
+    : Predicate that needs to succeed, otherwise `msg` is thrown
+
+    `msg`
+
+    : Message to throw in case `pred` fails
 
-     Type:
-       assertMsg :: Bool -> String -> Bool
   */
   # TODO(Profpatsch): add tests that check stderr
-  assertMsg =
-    # Predicate that needs to succeed, otherwise `msg` is thrown
-    pred:
-    # Message to throw in case `pred` fails
-    msg:
-    pred || builtins.throw msg;
-
-  /* Specialized `assertMsg` for checking if `val` is one of the elements
-     of the list `xs`. Useful for checking enums.
-
-     Example:
-       let sslLibrary = "libressl";
-       in assertOneOf "sslLibrary" sslLibrary [ "openssl" "bearssl" ]
-       stderr> error: sslLibrary must be one of [
-       stderr>   "openssl"
-       stderr>   "bearssl"
-       stderr> ], but is: "libressl"
-
-     Type:
-       assertOneOf :: String -> ComparableVal -> List ComparableVal -> Bool
+  assertMsg = pred: msg: pred || builtins.throw msg;
+
+  /**
+    Specialized `assertMsg` for checking if `val` is one of the elements
+    of the list `xs`. Useful for checking enums.
```

```suggestion
    Specialized `assertMsg` for checking if `val` is one of the elements of the list `xs`.
    Useful for checking enums.
```

I'm a bit conflicted on this, so do what makes more sense:
(applies to other functions too)
- The docs team prefers to use one line per sentence, because it makes it much easier to suggest changes (the GitHub interface for suggestions is based on lines).
- However, this is inside Nix code, so I understand there may be restrictions/style guides on column size or whatever, in which case I think these restrictions/styles have higher priority

(thinking about it, we have an RFC to get consistent styling for the entirely of Nixpkgs - @infinisil do you have any idea if this is something we should bring up on the Nixpkgs formatting RFC? I looked at their big haskell PR with the formatter implementation, and from the tests looks like they don't enforce anything on comments, but I couldn't tell)

---

Can't fix this really with this PR. since it is autogenerated.